src/service_info.rs

```diff
@@ -291,7 +291,7 @@ impl ServiceInfo {
     pub(crate) fn get_addrs_on_intf(&self, intf: &Interface) -> Vec<IpAddr> {
         self.addresses
             .iter()
-            .filter(|a| valid_ip_on_intf(a, intf))
+            .filter(|a| (a.is_loopback() || valid_ip_on_intf(a, intf)))
```

Comment:
I assume this is to support registering a service with a loopback address (maybe in addition to other addresses)? Seem OK to me.  Could you please add a comment here, and a test case? 

And I'm also curious why it's needed: currently this lib can already find local responders via the (non-loopback) IP address.